app/models/ability.rb

```diff
@@ -0,0 +1,39 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # guest user (not logged in)
+    if user.role == 'admin'
+      can :manage, :all
+    else
+      can :manage, Comment, author_id: user.id
+      can :manage, Post, author_id: user.id
+      can :read, :all
+    end
+
+    # Define abilities for the user here. For example:
+    #
+    #   return unless user.present?
```

Comment:
- [ optional] it will be fabulous if your create create your `user.admin?` method inside the user model.
eg 
```ruby
...
   def admin?
     role == 'admin' ? true : false
   end
```